tests/integration/test_fim/test_registry/test_recursion_level/test_recursion_level_registry.py

```diff
@@ -22,152 +19,134 @@
-
-        # Check False (exceeding the specified recursion_level)
-        for n in range(recursion_level, recursion_level + ignored_levels):
-            path = os.path.join(path, subkey + str(n + 1))
-            registry_value_cud(path, wazuh_log_monitor, time_travel=is_scheduled, min_timeout=timeout,
-                               triggers_event=False)
-
-    except TimeoutError:
-        timeout_log_monitor = FileMonitor(LOG_FILE_PATH)
-        if timeout_log_monitor.start(timeout=5, callback=callback_audit_event_too_long).result():
-            pytest.fail("Audit raised 'Event Too Long' message.")
-        raise
-
-    except FileNotFoundError as ex:
-        MAX_PATH_LENGTH_WINDOWS_ERROR = 206
-        if ex.winerror != MAX_PATH_LENGTH_WINDOWS_ERROR:
-            raise
-
-    except OSError as ex:
-        MAX_PATH_LENGTH_MACOS_ERROR = 63
-        MAX_PATH_LENGTH_SOLARIS_ERROR = 78
-        if ex.errno not in (MAX_PATH_LENGTH_SOLARIS_ERROR, MAX_PATH_LENGTH_MACOS_ERROR):
-            raise
-
-
-# Fixtures
-
-@pytest.fixture(scope='module', params=configurations)
-def get_configuration(request):
-    return request.param
-
-
-# Tests
-
-@pytest.mark.parametrize('key, registry, subkey, recursion_level', [
-    (key, registry_no_recursion, 'subkey', 0),
-    (key, registry_recrusion_1, 'subkey', 1),
-    (key, registry_recrusion_5, 'subkey', 5),
-    (key, registry_recrusion_512, 'subkey', 512)
-])
-def test_recursion_level(key, registry, subkey, recursion_level, get_configuration, configure_environment,
-                         restart_syscheckd, wait_for_initial_scan):
-    """
-    Check if files are correctly detected by syscheck with recursion level using scheduled, realtime and whodata
-    monitoring.
-
-    Parameters
-    ----------
-    key : str
-        String with the key of the registry (HKEY_* constants)
-    registry : str
-        String with the registry that is monitored (without the key string).
-    subkey : str
-        Name that will be used to generate subkeys int the monitored registry
-    recursion_level : int
-        Recursion level. Also used as the number of subkeys to be created and checked for the current test.
     """
-    recursion_test(key, registry, subkey, recursion_level, timeout=global_parameters.default_timeout,
-                   is_scheduled=get_configuration['metadata']['fim_mode'] == 'scheduled')
+    recursion_level = int(rl_dict[registry])
+    path = registry
+
+    # Check events in recursion level = 0
+    registry_value_cud(root_key, path, wazuh_log_monitor, arch=arch, time_travel=True,
+                       min_timeout=global_parameters.default_timeout)
+
+    path_list = list()
+    # For recursion lower levels, execute registry_value_cud in every level.
+    if recursion_level < edge_limit:
+        for level in range(recursion_level):
+            path = os.path.join(path, str(level + 1))
+            path_list.append(path)
+    else:
+        for level in range (recursion_level):
```

Comment:
Remove the whitespace before the `(`.
```suggestion
        for level in range(recursion_level):
```